Issue #2436:  NeedBraces: allow no body loops

MEZk opened the pull request.

romani commented:
We can not update ITs of Google style , It is strictly forbidden in Google Java Style.
http://checkstyle.sourceforge.net/reports/google-java-style.html#s4.1.1-braces-always-used

to reconcile requests - new option need to be create, default value is allowEmptyLoopBody=false.

MEZk commented:
@romani 
>typo: add space.

Removed the sentence. Moved to examples section.

>Please update Javdoc with the same content.

Done.

>to reconcile requests - new option need to be create, default value is allowEmptyLoopBody=false.

Done. Add UTs for allowEmptyLoopBody=false and allowEmptyLoopBody=true.

romani commented:
merged as FF.
